### PR TITLE
UI Slider - Add Thumb "always", ticks & vertical sliders

### LINKED ISSUE #372: Slider "Show thumb label" addition
### Description

It appears that the underlying vue.js control for the slider allows for a boolean OR `always` for `thumb-label`. I would love to be able to set it to this third value so the slider is ALWAYS displaying the current value. I would assume this would require changing the "Show thumb labels" setting to a dropdown or radio for the third value?

### Have you provided an initial effort estimate for this issue?

I can not provide an initial effort estimate

### LINKED ISSUE #624: UI-Slider orientation
### Description

Hello,

Is there any way to change the orientation of a UI slider from horizontal to vertical? In the base dashboard, there was an option for the slider to automatically switch to vertical orientation if its height was greater than its width.

### Have you provided an initial effort estimate for this issue?

I am no FlowFuse team member

### PR DESCRIPTION
**Pull request by joepavitt**
### Always Show Thumb

Screenshot 2024-06-23 at 08 46 55

### Tick Demo

Screenshot 2024-06-23 at 08 46 17

### Vertical Sliders:

Screenshot 2024-06-23 at 08 39 46

## Related Issue(s)

Closes #372 
Closes #624

### CODE REVIEW

**Comment by github-cyprien** [comment]
Thank you very much ! I've tried to add vertical option last week, but it wasn't looking good...
I will send you a PR (Hopefully today) to ad icons to slider.

File: nodes/widgets/locales/en-US/ui_slider.html [hunk 1]
```diff
@@ -6,16 +6,21 @@
         Each value change will generate a message with the value set as <b>payload</b>.
     </p>
     <p>The slider can be reversed by setting the min value larger than the max value. e.g. min 100, max 0.</p>
+    <p>Sliders will automatically render vertically if the <b>height</b> is set to be larger than the <b>width</b>.</p>
     <h3>Dynamic Properties (Inputs)</h3>
     <p>Any of the following can be appended to a <code>msg.ui_update.</code> in order to override or set properties on this node at runtime.</p>
     <dl class="message-properties">
         <dt class="optional">label <span class="property-type">string</span></dt>
         <dd>Add a CSS class, or more, to the Button at runtime.</dd>
     </dl>
```
Comment (Steve-Mcl) [inline nodes/widgets/locales/en-US/ui_slider.html:15]:
```suggestion
        <dt class="optional">label <span class="property-type">string</span></dt>
        <dd>Update the sliders label at runtime.</dd>
    </dl>
```

File: ui/src/widgets/ui-slider/UISlider.vue [hunk 2]
```diff
@@ -4,9 +4,10 @@
     <v-slider
         v-model="value" :disabled="!state.enabled" :label="label" hide-details="auto"
         :class="className" :thumb-label="thumbLabel"
-        :min="min"
+        :min="min" :direction="direction"
         :color="color" :track-color="colorTrack" :thumb-color="colorThumb"
-        :max="max" :step="props.step || 1" @update:model-value="onChange" @end="onBlur"
+        :max="max" :step="props.step || 1" :show-ticks="showTicks" @update:model-value="onChange"
```
Comment (Steve-Mcl) [inline ui/src/widgets/ui-slider/UISlider.vue:9]: Dynamic `step` value does not work
(Still bound to props.setp)

File: nodes/widgets/ui_slider.html [hunk 3]
```diff
@@ -70,15 +71,38 @@
...
+                // handle older iteration of thumbLabel
+                // otherwise, the value will match up to a valid selection
+                if (this.thumbLabel === false) {
+                    $('#node-input-thumbLabel').val('false')
+                } else if (this.thumbLabel === true) {
+                    $('#node-input-thumbLabel').val('true')
+                }
+
+                // handle no value for showTicks (from sliders created before this was a value)
+                console.log(this.showTicks)
```
Comment (Steve-Mcl) [inline nodes/widgets/ui_slider.html:83]:
```suggestion
```

**Review by Steve-Mcl — COMMENTED** [review]
Couple of minor issues and one visual thing, otherwise all good.

image

**Review by Steve-Mcl — APPROVED** [review]

**Pull request merged**